test/app-luatest/version_module_test.lua

```diff
@@ -0,0 +1,300 @@
+        {
+            str = '1.2.4-1',
+            ver = version.new(1, 2, 4, nil, 1),
+        },
+        {
+            str = '1.2.4-2',
+            ver = version.new(1, 2, 4, nil, 2),
+        },
+        {
+            str = '1.2.5-entrypoint',
+            ver = version.new(1, 2, 5, 'entrypoint', 0),
+        },
+        {
+            str = '1.2.5-entrypoint1-45-gc9981a567',
+            ver = version.new(1, 2, 5, 'entrypoint1', 45),
+        },
+        {
+            str = '1.2.5-alpha',
+            ver = version.new(1, 2, 5, 'alpha'),
+        },
+        {
+            str = '1.2.5-alpha1-45-gc9981a567',
+            ver = version.new(1, 2, 5, 'alpha1', 45),
+        },
+        {
+            str = '1.2.6-',
+            ver = version.new(1, 2, 6),
+        },
+        {
+            str = '1.2.7-entrypoint-',
+            ver = version.new(1, 2, 7, 'entrypoint'),
+        },
+        {
+            str = '1.2.7-entrypoint1-',
+            ver = version.new(1, 2, 7, 'entrypoint1'),
+        },
+        {
+            str = '1.2.7-entrypoint1-45',
+            ver = version.new(1, 2, 7, 'entrypoint1', 45),
+        },
+        {
+            str = '1.2.7-entrypoint1-46-',
+            ver = version.new(1, 2, 7, 'entrypoint1', 46),
+        },
+        {
+            str = '1.2.7-alpha-',
+            ver = version.new(1, 2, 7, 'alpha')
+        },
+        {
+            str = '1.2.7-alpha1-',
+            ver = version.new(1, 2, 7, 'alpha1')
+        },
+        {
+            str = '1.2.7-alpha1-45',
+            ver = version.new(1, 2, 7, 'alpha1', 45),
+        },
+        {
+            str = '1.2.7-alpha1-46-',
+            ver = version.new(1, 2, 7, 'alpha1', 46),
+        },
+        {
+            str = '1.2.8-entrypoint',
+            ver = version.new(1, 2, 8, 'entrypoint'),
+        },
+        {
+            str = '1.2.8-alpha',
+            ver = version.new(1, 2, 8, 'alpha'),
+        },
+        {
+            str = '1.2.8-beta',
+            ver = version.new(1, 2, 8, 'beta'),
+        },
+        {
+            str = '1.2.8-rc',
+            ver = version.new(1, 2, 8, 'rc'),
+        },
+        {
+            str = '1.2.8',
+            ver = version.new(1, 2, 8),
+        },
+    }
+    for i, v in pairs(versions) do
+        local ver = version.fromstr(v.str)
+        assert_version_str_equals(tostring(ver), v.str)
+
+        t.assert(ver == v.ver, ('%d'):format(i))
+        t.assert(ver == v.str, ('%d'):format(i))
+        t.assert(v.str == ver, ('%d'):format(i))
+        t.assert(v.ver == ver, ('%d'):format(i))
+
+        t.assert(ver <= v.ver, ('%d'):format(i))
+        t.assert(ver <= v.str, ('%d'):format(i))
+        t.assert(v.str <= ver, ('%d'):format(i))
+        t.assert(v.ver <= ver, ('%d'):format(i))
+
+        t.assert(ver >= v.ver, ('%d'):format(i))
+        t.assert(ver >= v.str, ('%d'):format(i))
+        t.assert(v.str >= ver, ('%d'):format(i))
+        t.assert(v.ver >= ver, ('%d'):format(i))
```

Comment:
Please use `t.assert_equals`, `t.assert_lt`, `t.assert_ge`, etc instead of comparing the result in the assertion. They yield a more verbose output in case of a failure.

Reply:
@locker I intentionally used `t.assert` and not its variations, as I wanna be sure, that luatest doesn't do some smart stuff (like for example for comparison of the tables [here](https://github.com/tarantool/luatest/blob/7dc5cb7d581312f48c97d8f394674975d5a45fd8/luatest/comparator.lua#L6-L25)). Moreover, `assert_lt/ge` accepts only numbers, see [here](https://github.com/tarantool/luatest/blob/7dc5cb7d581312f48c97d8f394674975d5a45fd8/luatest/assertions.lua#L365-L368), so comparison of cdata objects is not possible.

Reply:
AFAIR, I saw a discussion or a comment about it before. It seems worthful to include a note into the code as a comment, so this idea is clear for a reader and the question doesn't raise again.